packages/iris-grid/src/IrisGridUtils.ts

```diff
@@ -615,7 +622,7 @@ class IrisGridUtils {
 
   static dehydratePendingDataMap(
     columns: Column[],
-    pendingDataMap: Map<
+    pendingDataMap: ReadonlyMap<
       number,
       | UIRow
       | {
```

Comment:
This type seems a bit janky. I bet it could be something cleaner, but it's not part of these changes. I think `UIRow` can be removed and the data can just be `Map<ModelIndex, CellData | string | null>`. Then the extra casts/type definitions in the map function params can be removed

Reply:
Nice